src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs

```diff
@@ -153,5 +153,7 @@ public string GetLink<TAddress>(HttpContext httpContext, TAddress address, objec
             object values,
             LinkOptions options,
             out string link);
+
+        public abstract LinkGenerationTemplate GetTemplate(HttpContext httpContext, object values, LinkOptions options);
```

Comment:
A quick prototype based on my earlier api change. Here I put the `GetTemplate` api within LinkGenerator itself. I feel this is fine compared to having a separate factory. what do you guys think?

Reply:
This seems fine. This is a good thing for users, how it's implemented is up to us 👍